app/src/views/private/components/refresh-sidebar-detail.vue

```diff
@@ -35,16 +36,28 @@ export default defineComponent({
 
 		const activeInterval = ref<NodeJS.Timeout | null>(null);
 
+		const setRefreshInterval = (interval: number) => {
+			activeInterval.value = setInterval(() => {
+				emit('refresh');
+			}, interval * 1000);
+		};
+
+		idleTracker.on('hide', () => {
+			if (activeInterval.value) clearInterval(activeInterval.value);
+		});
+
+		idleTracker.on('show', () => {
```

Comment:
We should probably immediately emit a refresh when page becomes visible again, especially for lager intervals:

```suggestion
		idleTracker.on('show', () => {
			emit('refresh');
```

Reply:
> We should probably immediately emit a refresh when page becomes visible again, especially for lager intervals:

By doing this, we will face **one problem** i.e Token expiry error since auth token refresh is hitting after our `auto refresh` event. 
For example -> when  idle time is `1min` and token TTL is also `1min`. In this situation, we either have to handle the refresh auth token event first and our auto refresh event. 

Let me know whether to handle the events for this requirement ?

Reply:
Right, good point 👍 

Would be great if we could somehow await the [token refresh call](https://github.com/directus/directus/blob/e3e1f215733bd885146762722bcbe6c5b8464d36/app/src/auth.ts#L88) from within "auto refresh", but only refresh the token if actually necessary (see [`refreshIfExpired`](https://github.com/directus/sdk/blob/811950d8f3248ae636782c352875153c409859cf/src/base/auth.ts#L83) from SDK) so the token is not getting refreshed twice in a row.

Reply:
@paescuj I have updated the PR. Please have a look.